### PR TITLE
Unit Tests and ITs for Regex fix

### PR DESCRIPTION
**Pull request by allenhsieh**
Added unit tests and ITs for @adrianp-sti regex fix 

This pull request makes the following changes:
* Added unit test to check that if a package has a `.` in its' name that the fix will still parse the name and version correctly
* Added integration test to check that if a package has a `.` in its' name that the fix will still parse the name and version correctly

cc @DarthHater @bhamail

### CODE REVIEW

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java [hunk 1]
```diff
@@ -73,4 +73,24 @@ public void nameParseTest() throws Exception {
 
     assertThat(result, is(equalTo("a2ps-doc")));
   }
+
+  @Test
+public void nameWithDotParseTest() throws Exception {
```
Comment (DarthHater) [inline nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: FORMATTING: Indentation off

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java [hunk 2]
```diff
@@ -73,4 +73,24 @@ public void nameParseTest() throws Exception {
...
+  when(state.getTokens()).thenReturn(tokens);
+  when(tokens.get("filename")).thenReturn("lua5.3-libs-5.3.5-r2");
+
+  String result = underTest.name(state);
```
Comment (DarthHater) [inline nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: Technically you could inline this in the assertThat() so like: `assertThat(underTest.name(state), is(equalTo("lua5.3-libs")));`

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java [hunk 3]
```diff
@@ -73,4 +73,24 @@ public void nameParseTest() throws Exception {
...
+
+  assertThat(result, is(equalTo("lua5.3-libs")));
+}
+
+  @Test
+  public void versionWithDotParseTest() throws Exception {
+    when(state.getTokens()).thenReturn(tokens);
+    when(tokens.get("filename")).thenReturn("lua5.3-libs-5.3.5-r2");
+
+    String result = underTest.version(state);
```
Comment (DarthHater) [inline nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: Second verse, same as the first (you can inline this)

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 1 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 2 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 3 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 1 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9chttps://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 2 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

File: nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java (hunk 3 above)
Comment (allenhsieh) [reply nexus-repository-apk/src/test/java/org/sonatype/nexus/plugins/apk/internal/ApkPathUtilsTest.java]: in: https://github.com/adrianp-sti/nexus-repository-apk/pull/1/commits/140bd90e5da5d6dbf3c712a84599fd21f3547d9c

**Pull request merged**